### PR TITLE
Restore setting up temp dir for windows service

### PR DESCRIPTION
**Pull request by rjernst**
In https://github.com/elastic/elasticsearch/pull/41913 setting up the
temp dir for ES was moved from the env script to individual cli scripts.
However, moving it to the windows service cli was missed. This commit
restores setting up the temp dir for the windows service control script.

### CODE REVIEW

**Comment by rjernst** [comment]
Note I currently have the fix commented out here in order to ensure the new test would have caught the existing bug (I have had issues running the windows vagrant image on my local machine lately).

**Comment by rjernst** [comment]
Ok, this is ready for review. Getting tests running again required a couple fixes to how the java paths are handled, and I also added future protection in case our windows script fails.

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

File: distribution/src/bin/elasticsearch-service.bat [hunk 1]
```diff
@@ -106,6 +106,10 @@ if exist "%JAVA_HOME%\bin\server\jvm.dll" (
 )
 
 :foundJVM
+if not defined ES_TMPDIR (
+  for /f "tokens=* usebackq" %%a in (`CALL %JAVA% -cp "!ES_CLASSPATH!" "org.elasticsearch.tools.launchers.TempDirectory"`) do set  ES_TMPDIR=%%a
```
Comment (jasontedor) [inline distribution/src/bin/elasticsearch-service.bat]:
```suggestion
  for /f "tokens=* usebackq" %%a in (`CALL %JAVA% -cp "!ES_CLASSPATH!" "org.elasticsearch.tools.launchers.TempDirectory"`) do set ES_TMPDIR=%%a
```

**Review by jasontedor — APPROVED** [review]
LGTM.

I tested this in a Windows VM. I was able to reproduce the existing issue, and verify that this fix addresses the issue.

**Pull request merged**